packages/rancher-prom-to-teams/generated-changes/patch/templates/configmap.yaml.patch

```diff
@@ -0,0 +1,133 @@
+--- charts-original/templates/configmap.yaml
++++ charts/templates/configmap.yaml
+@@ -8,13 +8,13 @@
+ {{- if and (not .Values.prom2teams.connector) (not .Values.prom2teams.connectors) -}}
+ {{- fail "Invalid configuration: At least one connector must be provided"}}
+ {{- end -}}
++{{- $namespace := (include "prom2teams.namespace" . ) -}}
++{{- if and (.Release.IsInstall) (not (lookup "v1" "ConfigMap" $namespace (include "prom2teams.fullname" .))) }}
+ apiVersion: v1
+ kind: ConfigMap
+ metadata:
+-  name: prom2teams-config
+-  labels:
+-    app.kubernetes.io/managed-by: {{ .Release.Service }}
+-    app.kubernetes.io/instance: {{ .Release.Name }}
++  namespace: {{ $namespace }}
++  {{- include "prom2teams.preNameLabelAnno" . | nindent 2 }}
+ data:
+   config.ini: |-
+     [HTTP Server]
+@@ -33,4 +33,110 @@
+     Level: {{ .Values.prom2teams.loglevel }}
+     [Template]
+     Path: {{ .Values.prom2teams.templatepath }}
+-  teams.j2: {{ .Files.Get "files/teams.j2" |  quote }}
+\ No newline at end of file
++  teams.j2: {{ .Files.Get "files/teams.j2" |  quote }}
++---
++apiVersion: batch/v1
++kind: Job
++metadata:
++  namespace: {{ $namespace }}
++  {{- include "prom2teams.preNameLabel" . | nindent 2 }}
++  annotations:
++    "helm.sh/hook": pre-install
++    "helm.sh/hook-delete-policy": hook-succeeded,hook-failed,before-hook-creation
++    "helm.sh/hook-weight": "5"
++spec:
++  template:
++    metadata:
++    {{- include "prom2teams.preNameLabel" . | nindent 6 }}
++    spec:
++      serviceAccountName: {{ include "prom2teams.fullname" . }}-pre-install
++      {{- if .Values.global.preInstallHook.securityContext }}
++      securityContext:
++      {{- toYaml .Values.global.preInstallHook.securityContext | nindent 8 }}
++      {{- end }}
++      containers:
++        - name: copy-pre-install-configmap
++          image: {{ include "system_default_registry" . }}{{ .Values.global.preInstallHook.image.repository }}:{{ .Values.global.preInstallHook.image.tag }}
++          imagePullPolicy: {{ .Values.global.preInstallHook.image.pullPolicy }}
++          command:
++            - /bin/sh
++            - -c
++            - >
++              if kubectl get configmap -n {{ $namespace }} {{ include "prom2teams.fullname" . }} > /dev/null 2>&1; then
++                echo "configMap already exists"
++                exit 1
++              fi;
++              kubectl patch configmap -n {{ $namespace }} --dry-run -o yaml {{ include "prom2teams.preName" . }}
++              -p '{{ printf "{\"metadata\":{\"name\": \"%s\"}}" (include "prom2teams.fullname" .) }}'
++              | kubectl apply -f -;
++              kubectl annotate configmap -n {{ $namespace }} {{ include "prom2teams.preName" . }}
++              helm.sh/hook- helm.sh/hook-delete-policy- helm.sh/hook-weight-;
```

Comment:
Since this is fairly non-trivial, add this to a bash file in files/pre-install-job.sh and reference it as `{{ .Files.Get "files/pre-install-job.sh" }}` to embed it here

Reply:
A script was added.